### ISSUE TITLE
[feature request] Wayland support

### ISSUE THREAD

**Issue by MiguelDeLillo** [esjeon/krohnkite#49]
I tried using Kröhnkite in Wayland, but without any success. The session starts, but if Kröhnkite is enabled at boot all I see is the frozen cursor on a black background and I have to shut the PC down. 
The inability to use Kröhnkite is the only thing preventing me to switch to Wayland, as I could not even change the wm to a tiling one in it. It would be very nice of you to include support for Wayland, if it be possible.

**Comment by esjeon** [comment]
Last time I check, KWin intentionally broke the scripting interface. But it seems like they made a positive change some time ago. I'll take a look into it, though it would be of lower priority. It's a shame, but I have backlogs to catch up.

**Comment by MiguelDeLillo** [comment]
I appreciate your answer. Your extension is the only true way into KDE tiling: Wayland is soon to be ready, but I won't switch to it until you manage to port your extension.

**Comment by phrxmd** [comment]
This would be very welcome indeed! Missing KDE tiling is one of the Wayland showstoppers for me.

**Comment by phrxmd** [comment]
KWin scripting is now working in Wayland.
To test (tested with Plasma 5.19.5):

1. In the KDE Wayland session, open WM Console
2. Enter the following script:
````
    workspace.clientAdded.connect(function() { print("added"); })
    workspace.clientRemoved.connect(function() { print("removed"); })
````
3. Execute
4. Open and close a few windows and look at the console output.

**Comment by Ramblurr** [comment]
@phrxmd does this mean Krohnkite is working in wayland now?

**Comment by phrxmd** [comment]
@Ramblurr no, it isn't, it just means that KWin has the infrastructure in place to do it. I think it doesn't need lot of work, but someone with Typescript skills and a Wayland system needs to do it.

**Comment by oshanz** [comment]
#140
